7_deploy_osh/roles/suse-osh-deploy/files/wait-for-pods.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Copyright 2017 The Openstack-Helm Authors.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+set -e
+
+# From Kolla-Kubernetes, orginal authors Kevin Fox & Serguei Bezverkhi
+# Default wait timeout is 600 seconds
+end=$(date +%s)
+if ! [ -z $2 ]; then
+ end=$((end + $2))
+else
+ end=$((end + 100))
+fi
+while true; do
+    kubectl get pods --namespace=$1 -o json | jq -r \
+        '.items[].status.phase' | grep Pending > /dev/null && \
+        PENDING=True || PENDING=False
+    query='.items[]|select(.status.phase=="Running")'
+    query="$query|.status.containerStatuses[].ready"
+    kubectl get pods --namespace=$1 -o json | jq -r "$query" | \
+        grep false > /dev/null && READY="False" || READY="True"
+    kubectl get jobs -o json --namespace=$1 | jq -r \
+        '.items[] | .spec.completions == .status.succeeded or status == {}' | \
```

Comment:
Is the main difference here? Is this something we could propose upstream?

I would really appreciate if we could tag all borrowed files with their origins, along with comments as to what has been modified, and link(s) to relevant upstream reviews. If we need to update these later from what is upstream it'll be very helpful information to have.

Reply:
Will add comment about change. Correctness of this change needs to be validated as I cannot find in what case status : {} comes. I saw this in my local setup and added logic.
Upstream has this https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#jobstatus-v1-batch though not still no clue when empty value comes.